synapse/storage/push_rule.py

```diff
@@ -150,16 +150,11 @@ def _bulk_get_push_rules_for_room(self, room_id, state_group, current_state_ids,
         # their unread countss are correct in the event stream, but to avoid
         # generating them for bot / AS users etc, we only do so for people who've
         # sent a read receipt into the room.
-        local_user_member_ids = [
-            e_id for (etype, state_key), e_id in current_state_ids.iteritems()
-            if etype == EventTypes.Member and self.hs.is_mine_id(state_key)
-        ]
 
-        local_member_events = yield self._get_events(local_user_member_ids)
-
-        local_users_in_room = set(
-            member_event.state_key for member_event in local_member_events
-            if member_event.membership == Membership.JOIN
+        local_users_in_room = yield self._get_joined_users_from_context(
+            room_id, state_group, current_state_ids,
+            on_invalidate=cache_context.invalidate,
+            event=event,
```

Comment:
Oh, point. Although it'd get filtered out by the `get_if_users_have_pushers` below